api/api/src/main/scala/uk/ac/wellcome/platform/api/Responses.scala

```diff
@@ -0,0 +1,111 @@
+package uk.ac.wellcome.platform.api
+
+import io.circe.generic.extras.semiauto.deriveEncoder
+import io.circe.generic.extras.JsonKey
+import io.circe.{Encoder, Json}
+import akka.http.scaladsl.model.Uri
+
+import uk.ac.wellcome.platform.api.models._
+import uk.ac.wellcome.display.models._
+import uk.ac.wellcome.display.models.v2._
+import uk.ac.wellcome.display.models.Implicits._
+import uk.ac.wellcome.platform.api.services.WorksSearchOptions
+
+import io.circe.generic.extras.semiauto._
+import uk.ac.wellcome.display.json.DisplayJsonUtil._
+
+case class ResultResponse[T: Encoder](
+  @JsonKey("@context") context: String,
+  result: T
+)
+
+object ResultResponse {
+
+  // Flattens the 'result' field into the rest of the object
+  implicit def encoder[T: Encoder]: Encoder[ResultResponse[T]] =
+    deriveEncoder[ResultResponse[T]].mapJson { json =>
+      json.asObject
+        .flatMap { obj =>
+          obj.toMap
+            .get("result")
+            .flatMap(_.asObject.map(_.toList))
+            .map { fields =>
+              Json.obj(fields ++ obj.filterKeys(_ != "result").toList: _*)
+            }
+        }
+        .getOrElse(json)
+    }
+}
+
+case class MultipleWorksResponse(
+  @JsonKey("@context") context: String,
```

Comment:
Could we somehow make this a static value, as it is always the same? I've given it a go in the old world and couldn't, but I actually think that might be a circe thing.

Reply:
I have updated it so we just calculate the request URI once when we load up the router, rather than per request. See [here](https://github.com/wellcometrust/catalogue/pull/192/commits/b794660969c4ba52d3e7e8266ed37ace65e734ee).

Reply:
(We can't make it completely static as is dependent on the `ApiConfig`, which isn't static and is generated from configuration)